Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM ubuntu/postgres:12-20.04_beta
+
+RUN echo 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend \
+    && apt-get update -y \
+    # Install Patroni and its dependencies.
+    && apt-cache depends patroni | sed -n -e 's/.* Depends: \(python3-.\+\)$/\1/p' \
+            | grep -Ev '^python3-(sphinx|etcd|consul|kazoo|kubernetes)' \
+            | xargs apt-get install -y git python3-pip python3-wheel \
+    && pip3 install setuptools \
+    && pip3 install 'git+https://github.com/zalando/patroni.git@v2.1.2#egg=patroni[kubernetes]' \
+    # Clean up.
+    && apt-get remove -y git python3-pip python3-wheel \
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/* /root/.cache
+
```

Comment:
Should we include a non-root user in here and switch to it by default? It'd also be a good idea to have a default entrypoint that does something sensible, even if we're going to override with Pebble.

Reply:
I agree with the point about the user. It makes sense to switch to the postgres user, which owns patroni and postgresql processes. I'll add it.

Reply:
That sounds good - just a simple scripts that either writes the contents of env files to the right places, does the basic setup and starts patroni - in the simplest way possible :)

Reply:
Default user and basic entrypoint were created.

Reply:
I realized that the docker image is specific to k8s (it doesn't run on docker because Patroni is installed with dependencies related to k8s). So I created an k8s manifest in order to enable the user to use this image without a charm.